src/test/java/htsjdk/samtools/SamReaderFactoryTest.java

```diff
@@ -284,7 +284,37 @@ public void queryInputResourcePermutation(final SamInputResource resource) throw
         }
         reader.close();
     }
-    
+
+    public class NeverFilePathInpurResource extends PathInputResource {
```

Comment:
This class should definitely be `private static`, and come with a comment explaining what it's for.

Also, typo: `Inpur` -> `Input`